chain/solana/solana.go

```diff
@@ -29,22 +54,83 @@ func DefaultClientOptions() ClientOptions {
 	}
 }
 
+// WithRPCURL returns a modified version of the options with the given API
+// rpc-url
+func (opts ClientOptions) WithRPCURL(rpcURL pack.String) ClientOptions {
+	opts.RPCURL = string(rpcURL)
+	return opts
+}
+
+// Client represents a Solana client that implements the multichain Contract API.
 type Client struct {
 	opts ClientOptions
 }
 
+// NewClient returns a new solana.Client interface that implements the
+// multichain Contract API.
 func NewClient(opts ClientOptions) *Client {
 	return &Client{opts: opts}
 }
 
-func (client *Client) CallContract(ctx context.Context, contract address.Address, input pack.Bytes) (output pack.Bytes, err error) {
-	if input != nil && len(input) != 0 {
-		return nil, fmt.Errorf("expected nil input, got %v input", input)
+// FindProgramAddress is a wrapper function that calls the Solana FFI to find
+// the deterministic program-derived address using the program and seeds.
+func FindProgramAddress(seeds []byte, program address.RawAddress) (address.Address, error) {
+	addrEncodeDecoder := NewAddressEncodeDecoder()
+	encoded, err := addrEncodeDecoder.EncodeAddress(program)
+	if err != nil {
+		return address.Address(""), err
+	}
+
+	return ProgramDerivedAddress(seeds, encoded), nil
+}
+
+// BurnCallContractInput represents the data structure of the expected calldata
+// for Solana's Contract API implementation.
+type BurnCallContractInput struct {
```

Comment:
I think this should not be defined here. It is application-specific, and so it should go in the Darknode.